lib/AST/Expr.cpp

```diff
@@ -57,6 +57,30 @@ void *Expr::operator new(size_t Bytes, ASTContext &C,
   return C.Allocate(Bytes, Alignment);
 }
 
+DefaultArgumentKind Expr::getDefaultArgKind(Expr *expr) {
```

Comment:
Should this operate on `this` instead of a parameter?

Reply:
I initially did it that way, but in ParsePattern I had to write something like:
```cpp
if (param.DefaultArg) {
  kind = param.DefaultArg->getDefaultArgKind();
} else {
  kind = DefaultArgumentKind::None;
}
```
This isn't a huge deal, but it certainly seemed nicer to just write `Expr::getDefaultArgKind(param.DefaultArg);` instead.

Reply:
I personally prefer the former, it makes it slightly more clear what you're doing. Also you could just write

```c++
auto kind = param.DefaultArg ?
            param.DefaultArg->getDefaultArgKind :: DefaultArgKind::None;
```